### ISSUE TITLE
Driver Pod fails on Azure Stack Hub because Key Vault Secret URI is incomplete

### ISSUE THREAD

**Issue by seb0006** [Azure/secrets-store-csi-driver-provider-azure#504]
We have installed the driver on a Linux AKS cluster using kubernetes version 1.16.9 on Azure Stack Hub (we are using AKS Private Preview) and have found that the driver pod fails to start with a "404 resource not found" error.

![404_pod_error](https://user-images.githubusercontent.com/49771811/117858939-68357000-b25c-11eb-8428-01921938fa3a.PNG)

We've tried 2 different configurations on 2 separate AKS clusters:
1. "keyVaultEndpoint":"https://vault.adfs.spokane.selfhost.local/a359277c-f0a8-4d2f-b22a-4591229629bs"
2. "keyVaultEndpoint":"https://vault.spokane.azurestack.corp.microsoft.com"

which both fail with a 404.

Here's some additional information comparing Azure Key Vault in the Public Cloud (left) and its sample secret KV URI (reachable) vs ADFS Azure Stack Hub KV and its sample secret KV URI (unreachable):

![kv_resource_uri_comparison](https://user-images.githubusercontent.com/49771811/117859482-0cb7b200-b25d-11eb-9a2e-db0ae40d6848.png)

**Comment by aramase** [comment]
@seb0006 thank you! Please let us know once you know more.

**Comment by aramase** [comment]
@seb0006 Any updates on this? Is the issue resolve now? I would love to know if there is anything we're missing in the docs.

**Comment by seb0006** [comment]
@aramase - I apologize for the delayed response.  The Azure Stack Hub AKS product group has the CSI driver pre-installed when a new AKS cluster is provisioned with all of the correct setup required for use on AKS.  The issue here was caused by my team trying to do the installation manually. 

Nothing is missing in your documentation; there are just some differences between public Azure and Azure Stack Hub that we had to work through with the PG.
